internal/backends/backend_test.go

```diff
@@ -62,3 +62,91 @@ func TestVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestListDatabasesAndListCollections(t *testing.T) {
+	t.Parallel()
+
+	ctx := conninfo.Ctx(testutil.Ctx(t), conninfo.New())
+
+	for name, b := range testBackends(t) {
+		t.Log("testing backend: ", name)
+		name, b := name, b
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			// setup 3 DB with 1 ,1 and 3 collections respectively
+			dbNames := []string{"testDB2", "testDB1", "testDB3"}
+			collectionNames := []string{"testCollection2", "testCollection1", "testCollection3"}
+
+			testDB, err := b.Database(dbNames[0])
+			require.NoError(t, err)
+			err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionNames[0]})
+			require.NoError(t, err)
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[0]})
+
+			testDB, err = b.Database(dbNames[1])
+			require.NoError(t, err)
+			err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionNames[0]})
+			require.NoError(t, err)
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[1]})
+
+			testDB, err = b.Database(dbNames[2])
+			require.NoError(t, err)
+			for _, collectionName := range collectionNames {
+				err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionName})
+				require.NoError(t, err)
+			}
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[2]})
+
+			t.Run("ListDatabases and ListCollctions test", func(t *testing.T) {
```

Comment:
We normally use CamelCase instead of "words with spaces" for subtests too

Reply:
done